crates/cheatnet/src/forking/cache.rs

```diff
@@ -0,0 +1,298 @@
+            file.read_to_string(&mut cache_file_content)
+                .expect("Could not read cache file: {path}");
+
+            // File was just created
+            let fork_cache_content = if cache_file_content.is_empty() {
+                ForkCacheContent::new()
+            } else {
+                ForkCacheContent::from_str(cache_file_content.as_str())
+            };
+
+            (fork_cache_content, Some(cache_file_path.to_string()))
+        };
+
+        ForkCache {
+            fork_cache_content,
+            cache_file,
+            block_id,
+        }
+    }
+
+    fn save(&self) {
+        let cache_file = self
+            .cache_file
+            .clone()
+            .unwrap_or_else(|| panic!("No cache_file to save to"));
+        let mut file = OpenOptions::new()
+            .write(true)
+            .create(true)
+            .open(cache_file.clone())
+            .unwrap();
+
+        file.lock_exclusive().expect("Could not lock on cache file");
+
+        let cache_file_content =
+            fs::read_to_string(cache_file).expect("Should have been able to read the cache");
+
+        let output = if cache_file_content.is_empty() {
+            self.fork_cache_content.to_string()
+        } else {
+            let mut fs_fork_cache_content = ForkCacheContent::from_str(cache_file_content.as_str());
+            fs_fork_cache_content.apply(&self.fork_cache_content);
+            fs_fork_cache_content.to_string()
+        };
+
+        file.write_all(output.as_bytes())
+            .expect("Could not write cache to file");
+
+        file.unlock().unwrap();
+    }
+
+    pub(crate) fn get_storage_at(
+        &self,
+        contract_address: ContractAddress,
+        key: StorageKey,
+    ) -> Option<StarkFelt> {
+        let contract_address_str = contract_address.to_felt252().to_string();
+        let storage_key_str = key.0.key().to_felt252().to_string();
+
+        let cache_hit = self
+            .fork_cache_content
+            .storage_at
+            .get(&contract_address_str)?
+            .get(&storage_key_str)?;
+
+        Some(
+            FieldElement::from_hex_be(cache_hit)
+                .unwrap_or_else(|_| panic!("Could not parse {cache_hit}"))
+                .to_stark_felt(),
+        )
+    }
+
+    pub(crate) fn cache_get_storage_at(
+        &mut self,
+        contract_address: ContractAddress,
+        key: StorageKey,
+        value: StarkFelt,
+    ) {
+        let contract_address_str = contract_address.to_felt252().to_string();
+        let storage_key_str = key.0.key().to_felt252().to_string();
+        let value_str = value.to_felt252().to_string();
+
+        if !self
+            .fork_cache_content
+            .storage_at
+            .contains_key(&contract_address_str)
+        {
+            self.fork_cache_content
+                .storage_at
+                .insert(contract_address_str.clone(), HashMap::new());
+        };
```

Comment:
This can be simplified with `entry.or_insert_with` https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#:~:text=//%20insert%20a%20key%20using%20a%20function%20that%20provides%20a%20new%20value%20only%20if%20it%0A//%20doesn%27t%20already%20exist

Reply:
So this would be smth like

```suggestion
        self.fork_cache_content
            .storage_at
            .entry(contract_address_str).or_insert_with(HashMap::new);
 
```